src/sentry/search/events/datasets/spans_indexed.py

```diff
@@ -735,3 +761,154 @@ def _resolve_percentile_weighted(
             [args["column"], Function("toUInt64", [self.sampling_weight])],
             alias,
         )
+
+    def _query_total_counts(self) -> tuple[float | int, float | int]:
+        if self._cached_count is None:
+            total_query = spans_indexed.SpansEAPQueryBuilder(
+                dataset=self.builder.dataset,
+                params={},
+                snuba_params=self.builder.params,
+                selected_columns=["count()", "count_weighted()"],
+            )
+            total_results = total_query.run_query(Referrer.API_SPANS_TOTAL_COUNT_FIELD.value)
+            results = total_query.process_results(total_results)
+            if len(results["data"]) != 1:
+                raise Exception("Could not query population size")
+            self._cached_count = results["data"][0]["count"]
+            self._cached_count_weighted = results["data"][0]["count_weighted"]
+        return self._cached_count, self._cached_count_weighted
+
+    def _resolve_margin_of_error(
+        self,
+        args: Mapping[str, str | Column | SelectType | int | float],
+        alias: str | None = None,
+    ) -> SelectType:
+        """Calculates the Margin of error for a given value, but unfortunately basis the total count based on
+        extrapolated data"""
+        # both of these need to be aggregated without a query
+        total_samples, population_size = self._query_total_counts()
+        sampled_group = Function("count", [])
+        return Function(
+            "multiply",
+            [
+                # Based on a z score for a confidence level of 95%
+                1.96,
+                Function(
+                    "multiply",
+                    [
+                        # Unadjusted Margin of Error
+                        self._resolve_unadjusted_margin(sampled_group, total_samples),
+                        # Finite Population Correction
+                        self._resolve_finite_population_correction(total_samples, population_size),
+                    ],
+                ),
+            ],
+            alias,
+        )
+
+    def _resolve_unadjusted_margin(
+        self, sampled_group: SelectType, total_samples: SelectType
+    ) -> SelectType:
+        # Naming this p to match the formula
+        p = Function("divide", [sampled_group, total_samples])
+        return Function(
+            "sqrt",
+            [
+                Function(
+                    "divide", [Function("multiply", [p, Function("minus", [1, p])]), total_samples]
+                )
+            ],
+        )
```

Comment:
I'm a little rusty on snql functions, do we have to worry about / 0 on any of these intermediate values? Or does everything chain / coerce nicely to null

Reply:
The current clickhouse version we're on will `inf` the `/0`, we're not showing this to users yet so i'll leave them for now, but i'll update this in the next version
```
select 1/0

 divide(1, 0)

          inf

```